code/src/main/java/com/onlinecontacttracing/messaging/MessageServlet.java

```diff
@@ -0,0 +1,97 @@
+import com.google.gson.Gson;
+import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import java.util.Collections;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.Projection;
+
+@WebServlet("/messaging")
+public class MessageServlet extends HttpServlet {
+  private SystemMessage systemMessage;
+  private LocalityResource localityResource;
+  private CustomizableMessage customizableMessage;
+  private String userMessage;
+  private PositiveUser user;
+  //for backend use
+  private String compiledMessage;
+  //for frontend display
+  private List<String> statusMessage = new ArrayList<String> ();
+
+
+  public MessageServlet(SystemMessage systemMessage, LocalityResource localityResource, CustomizableMessage customizableMessage, PositiveUser user) {
+    this.systemMessage = systemMessage;
+    this.localityResource = localityResource;
+    this.customizableMessage =  customizableMessage;
+    this.userMessage = customizableMessage.getMessage();
+    this.user = user;
+
+  }
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    String messageLanguage = request.getParameter("language");
+    response.getWriter().println(statusListToShowUser(messageLanguage));
+  }
+
+  public void checkForFlags() {
+    CheckMessagesForFlags flagChecker = new CheckMessagesForFlags();
+    statusMessage = CheckMessagesForFlags.findTriggeredFlags(flagChecker, user, userMessage);
+  }
+
+  //if flags are triggered, returns message and list of errors, else returns message
+  public List<String> statusListToShowUser(String messageLanguage) {
+    String translatedResourceMessage;
+    String translatedSystemMessage;
+    user.incrementAttemptedEmailDrafts();
+    checkForFlags();
+    
+    if (messageLanguage.equals("SP")) {
+    translatedResourceMessage = localityResource.getEnglishTranslation();
+    translatedSystemMessage = systemMessage.getEnglishTranslation();
+    }
+    else {
+      translatedResourceMessage = localityResource.getEnglishTranslation();
+      translatedSystemMessage = systemMessage.getEnglishTranslation();
+    }
+    
+    //if any flags are triggered, do not want to include the user's message in the message sent to the backend
+    if (statusMessage.size() != 0){
+      compiledMessage = translatedSystemMessage.concat(translatedResourceMessage);
+    } else {
+      compiledMessage = (translatedSystemMessage.concat(userMessage).concat(translatedResourceMessage);
+    }
+    statusMessage.add(translatedSystemMessage.concat(userMessage).concat(translatedResourceMessage));
```

Comment:
I note you don't do the check here as well. 

What is the compiledMessage being used as? It's a little concerning that we have these two fields that are almost exactly the same.

Reply:
The idea was that statusMessage is to be displayed to the user in the frontend, and compiledMessage will be used in the backend. For the frontend message, we still want to display the user's message to them even if there are errors, but for the backend, if there are flags, we wouldn't want to include the message in the email. But I think these variable names are probably not a descriptive as they could be so I'll fix that

Reply:
and also maybe add more comments to help clarify